sn_node/src/replication.rs

```diff
@@ -76,12 +79,27 @@ impl Node {
                         target_record: None,
                         expected_holders: Default::default(),
                     };
-                    match node.network().get_record_from_network(key, &get_cfg).await {
+                    match node
+                        .network()
+                        .get_record_from_network(key.clone(), &get_cfg)
+                        .await
+                    {
                         Ok(record) => record,
-                        Err(err) => {
-                            error!("During replication fetch of {pretty_key:?}, failed in re-attempt of get from network {err:?}");
-                            return;
-                        }
+                        Err(error) => match error {
+                            sn_networking::NetworkError::DoubleSpendAttempt(spends) => {
+                                debug!("Failed to fetch record {pretty_key:?} from the network, double spend attempt {spends:?}");
+
+                                let bytes = try_serialize_record(&spends, RecordKind::Spend)?;
+
+                                Record {
+                                    key,
+                                    value: bytes.to_vec(),
```

Comment:
this means seralize two spends into one Record?
the `try_serialize_records`, and the `get_spend_from_record` functions are all shall get refactored to accomodate this.

Also, say one reocord is having content [1,2], and the other is with content [1], then a get_record will result with `SplitRecord` with both, then how these two got merged ?

Reply:
> then how these two got merged ?

That's for node layer logic outwith of the get function now. (So the register apis eg, or the spend validation steps which already do this)

Reply:
> this means seralize two spends into one Record?

This is already how this is stored, so there should be no change here. We're just ensuring that we do this on replicaiton now.